### PR TITLE
Examples guides

### PR DESCRIPTION
**Pull request by danilo-pejovic**
LEDs: 

 - fixing order of LEDs
 - fixing incorrect LED topic name in led node 
 - adding led_test.py example where RAE flashes different leds based on movement direction
 
Motors: 
  - moving motors from counting revolutions to counting ticks
  - Moving calculate speed to control motors thread
  - Resetting I parameter in PID control every 500 ticks
  - Adjusting tests and ros2_control parameters to changes

### CODE REVIEW

File: rae_description/urdf/rae_ros2_control.urdf.xacro [hunk 1]
```diff
@@ -15,18 +15,18 @@
       <param name="enLB">43</param>
       <param name="enRA">46</param>
       <param name="enRB">47</param>
-      <param name="encTicsPerRevL">187</param>
-      <param name="encTicsPerRevR">187</param>
+      <param name="encTicsPerRevL">756</param>
```
Comment (Serafadam) [inline rae_description/urdf/rae_ros2_control.urdf.xacro:18]: Can you showcase the difference between old an new version?

File: rae_hw/src/rae_motors.cpp [hunk 2]
```diff
@@ -54,10 +54,10 @@ namespace rae_hw
         stop();
     }
 
-    void RaeMotor::calcSpeed()
+    float RaeMotor::calcSpeed()
     {
-        while (_running)
-        {
+        // while (_running)
```
Comment (Serafadam) [inline rae_hw/src/rae_motors.cpp]: Please remove the code instead of commenting it out

File: rae_hw/src/rae_motors.cpp [hunk 3]
```diff
@@ -78,27 +79,29 @@ namespace rae_hw
...
                 dir = (outSpeed >= 0) ^ reversePhPinLogic_;
                 prevErrorTime = currTime;
                 prevError = error;
-                if (dutyCycleFile.is_open())
-        {
-            dutyCycleFile << dutyCycle;
-            dutyCycleFile.close();
-        }
-            usleep(100000);
+                //std::cout << "CLS; MOTOR: " << pwmPin << "SPEED: " << outSpeed << "DC: "<< dutyCycle << std::endl;
```
Comment (Serafadam) [inline rae_hw/src/rae_motors.cpp]: Same here w.r.t comments

**Review by Serafadam — CHANGES_REQUESTED** [review]
When running `robot.launch.py`, following error gets printed out:
`Logical LED ID 39 doesn't have a corresponding physical LED.`

File: rae_description/urdf/rae_ros2_control.urdf.xacro (hunk 1 above)
Comment (danilo-pejovic) [reply rae_description/urdf/rae_ros2_control.urdf.xacro:18]: [You can check out how encoders are polled here](https://github.com/luxonis/rae-ros/blob/8e4a8a1d564e1cda91b14e674dc261494ba164c6/rae_hw/src/rae_motors.cpp#L190) - now they take count from more states leading to 4x better resolution [compared to previous version](https://github.com/luxonis/rae-ros/blob/057f2d05c5e803a692c5e8f5356f72dae8c908fa/rae_hw/src/rae_motors.cpp#L182). It does take more computing power but more responsive control should be worth it.

File: rae_hw/src/rae_motors.cpp [hunk 4]
```diff
@@ -183,57 +189,96 @@ namespace rae_hw
...
+                    count++;
+                    direction=true;
+                }
+                else if (prevState == Halfway)
+                {
+                    count--;
+                    direction=false;
+                }
+                else{
+                   // std::cout << "We are missing ticks, adding or removing 2 based on previous state  " << pwmPin << "Prev and current state" << prevState.A << prevState.B << currS.A << currS.B << std::endl;
```
Comment (Serafadam) [inline rae_hw/src/rae_motors.cpp]: There's few comments here as well

File: rae_hw/src/rae_motors.cpp [hunk 5]
```diff
@@ -287,7 +332,7 @@ namespace rae_hw
         _running = false;
         disablePWM();
         encoderThread.join();
-        calcSpeedThread.join();
+        // calcSpeedThread.join();
```
Comment (Serafadam) [inline rae_hw/src/rae_motors.cpp]: This should also be removed

File: rae_hw/src/rae_motors.cpp (hunk 4 above)
Comment (danilo-pejovic) [reply rae_hw/src/rae_motors.cpp]: Didnt git pull before pushing so git just ignored my changes for some reason

**Review by Serafadam — APPROVED** [review]
LGTM

**Pull request merged**